scrunch/mutable_dataset.py

```diff
@@ -0,0 +1,892 @@
+                "Please try again using its id")
+
+        body = dict(
+            dataset=fork_url,
+            autorollback=autorollback
+        )
+        self.resource.actions.post(body)
+
+    def delete_forks(self):
+        """
+        Deletes all the forks on the dataset. CANNOT BE UNDONE!
+        """
+
+        for fork in six.itervalues(self.resource.forks.index):
+            fork.entity.delete()
+
+    def join(self, left_var, right_ds, right_var, columns=None,
+             filter=None, wait=True):
+        """
+        Joins a given variable. In crunch joins are left joins, where
+        left is the dataset variable and right is other dataset variable.
+        For more information see:
+        http://docs.crunch.io/?http#merging-and-joining-datasets
+
+        :param: columns: Specify a list of variables from right dataset
+        to bring in the merge:
+        http://docs.crunch.io/?http#joining-a-subset-of-variables
+
+        :param: wait: Wait for the join progress to finish by polling
+        or simply return a url to the progress resource
+
+        :param: filter: Filters out rows based on the given expression,
+        or on a given url for an existing filter. TODO: for the moment
+        we only allow expressions
+        """
+        right_var_url = right_ds[right_var].url
+        left_var_url = self[left_var].url
+        # this dictionary sets the main part of the join
+        adapter = {
+            'function': 'adapt',
+            'args': [
+                {'dataset': right_ds.url},
+                {'variable': right_var_url},
+                {'variable': left_var_url}
+            ]
+        }
+
+        # wrap the adapter method on a shoji and body entity
+        payload = {
+            'element': 'shoji:entity',
+            'body': adapter
+        }
+
+        if columns and isinstance(columns, list):
+            # overwrite body to new format
+            payload['body'] = {
+                'frame': adapter,
+                'function': 'select',
+                'args': [
+                    {'map': {}}
+                ]
+            }
+            # add the individual variable columns to the payload
+            for var in columns:
+                var_url = right_ds[var].url
+                payload['body']['args'][0]['map'][var_url] = {'variable': var_url}
+
+        if filter:
+            # in the case of a filter, convert it to crunch
+            # and attach the filter to the payload
+            expr = process_expr(parse_expr(filter), right_ds)
+            payload['body']['filter'] = {'expression': expr}
+
+        progress = self.resource.variables.post(payload)
+        # poll for progress to finish or return the url to progress
+        if wait:
+            return wait_progress(r=progress, session=self.resource.session, entity=self)
+        return progress.json()['value']
+
+    def create_numeric(self, alias, name, derivation, description='', notes=''):
```

Comment:
This and create_categorical, since they create derived variables, can go on BaseVariable